### PR TITLE
[Android] Proto moved to separate module

### PR DESCRIPTION
**Pull request by MaximPestryakov**
## Description

- Moved generated proto classes to `:wallet-core-proto` module. So, now we will have two artifacts: `com.trustwallet:wallet-core` and `com.trustwallet:wallet-core-proto`. But you still can import just `com.trustwallet:wallet-core` and proto will come as transitive dependency
- Updated `com.google.protobuf:protobuf-javalite` runtime library to `3.22.3`
- Removed generation of useless `BuildConfig` and `R` classes

We need separate proto artifact to smoothly migrate to Wallet Core Kotlin. We want to keep old protos for the first step

### CODE REVIEW

File: android/wallet-core-proto/build.gradle [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apply plugin: 'java-library'
+apply plugin: 'maven-publish'
+
+group = 'com.trustwallet'
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (AndrewChupin) [inline android/wallet-core-proto/build.gradle:8]: Up to VERSION_11

**Review by AndrewChupin — DISMISSED** [review]

File: android/settings.gradle [hunk 2]
```diff
@@ -1 +1 @@
-include ':app', ':trustwalletcore'
+include ':app', ':trustwalletcore', ':wallet-core-proto'
```
Comment (hewigovens) [inline android/settings.gradle:1]: let's change `trustwalletcore` to `wallet-core`? too?

File: android/wallet-core-proto/maven-push.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    }
+                }
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            name = "GitHubPackages"
+            url = uri("https://maven.pkg.github.com/trustwallet/wallet-core")
```
Comment (hewigovens) [inline android/wallet-core-proto/maven-push.gradle:38]: Do we need to trigger release CI twice after this change?

Comment (MaximPestryakov) [reply android/wallet-core-proto/maven-push.gradle:38]: If you call `tools/android-release` then no. It assembles and publishes both modules

**Review by Milerius — CHANGES_REQUESTED** [review]
CI is not passing atm, please let's run the test locally before querying a review. Other than that looking good! thanks for the PR!

File: android/settings.gradle (hunk 2 above)
Comment (MaximPestryakov) [reply android/settings.gradle]: Fixed

**Review by Milerius — APPROVED** [review]

**Review by Milerius — APPROVED** [review]

**Pull request merged**